src/freadR.h

```diff
@@ -2,6 +2,7 @@
 #define dt_FREAD_R_H
 #define STRICT_R_HEADERS   // https://cran.r-project.org/doc/manuals/r-devel/R-exts.html#Error-handling
 #include <R.h>
+#include <Rinternals.h>
```

Comment:
I understand we need this for `translateChar(mkCharCE(s, ienc))`, right?
It may be good to ask R core for public API for this functionality.

Reply:
Maybe but I have the impression from WRE that Rinternals.h is already a public API, at least partially. For example, the string "#include <Rinternals.h>" is exhibited 13 times in WRE. We need it even for just SEXP (which is stated too in WRE). I did find one use of the word 'public' in the context of Rinternals.h too:
![image](https://user-images.githubusercontent.com/2779998/103565400-85f16380-4e7d-11eb-9413-e5f3cbe80b0a.png)
Note 144 in full is "see [The R API](https://cran.r-project.org/doc/manuals/r-devel/R-exts.html#The-R-API): note that these are not all part of the API."
I remember seeing some debate on R-devel mailing list on this over the years. Some R-core members believe that the R API is anything mentioned in WRE. But that would exclude many of the exported (i.e. non-hidden) functions that many packages use. I believe that `#define USE_RINTERNALS` before "#include <Rinternals.h>" (as data.table used to do in the past) does access truly internal non-API functions/structures. But including Rinternals.h is not necessarily accessing non-public API, in fact it's necessary to include it to access all the public API.
However, with that said, `translateChar` and `mkCharCE` are both highlighted by name in WRE (section 5.15) and for that reason there would be no disagreement by any R-core member that those are already public API.

Reply:
that seems... needlessly confusing!

Reply:
I agree. All those that have thought so have engaged. Then what you discover is that R-core members disagree between themselves. Tangent issues (such as data.table using SET_TRUELENGTH) are quickly raised. I assume what you would like is for R.h to be public and Rinternals.h to be not-public? Underlying that seemingly simple task is i) change to R over time, and ii) in our case of data.table, achievements by using SET_TRUELENGTH that some core members wish was never accessible.
Careful what you wish for.

Reply:
Yes, sorry for fuss. When I read Rinternals I had in mind USE_INTERNALS switch. So definitely confusing.